client/client.go

```diff
@@ -14,10 +14,18 @@ import (
 
 // Client defines the interface for interacting with a Git repository.
 type Client interface {
-	// TODO(mem): this is probably not the right interface.
-	SendCommands(ctx context.Context, data []byte) ([]byte, error)
-	SmartInfoRequest(ctx context.Context) ([]byte, error)
+	// TODO: this is probably not the right interface, we may have to separate the client
+	// Raw protocol requests
+	// UploadPack sends a POST request to the git-upload-pack endpoint.
+	UploadPack(ctx context.Context, data []byte) ([]byte, error)
+	// ReceivePack sends a POST request to the git-receive-pack endpoint.
+	ReceivePack(ctx context.Context, data []byte) ([]byte, error)
+	// SmartInfo sends a GET request to the info/refs endpoint.
+	SmartInfo(ctx context.Context, service string) ([]byte, error)
```

Comment:
these feel very internal... where the ones below are the external/useable

does it make sensse to expose the raw pack operations in the main interface?

Reply:
yeah, I am also thinking in moving the raw client to protocol package and keep the client package for the high-level operations.